modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/AbstractSnapshotVerificationTask.java

```diff
@@ -100,19 +98,60 @@ public abstract class AbstractSnapshotVerificationTask extends
 
     /**
      * @param name Snapshot name.
-     * @param path Snapshot directory path.
-     * @param incIdx Incremental snapshot index.
      * @param constId Snapshot metadata file name.
-     * @param groups Cache groups to be restored from the snapshot. May be empty if all cache groups are being restored.
-     * @param check If {@code true} check snapshot before restore.
+     * @param args Check snapshot parameters.
+     *
      * @return Compute job.
      */
-    protected abstract ComputeJob createJob(
-        String name,
-        @Nullable String path,
-        int incIdx,
-        String constId,
-        Collection<String> groups,
-        boolean check
-    );
+    protected abstract AbstractSnapshotPartitionsVerifyJob createJob(String name, String constId, SnapshotPartitionsVerifyTaskArg args);
+
+    /** */
+    protected abstract static class AbstractSnapshotPartitionsVerifyJob extends ComputeJobAdapter {
+        /** Serial version uid. */
+        private static final long serialVersionUID = 0L;
+
+        /** Ignite instance. */
+        @IgniteInstanceResource
+        protected IgniteEx ignite;
+
+        /** Injected logger. */
+        @LoggerResource
+        protected IgniteLogger log;
+
+        /** Snapshot name. */
+        protected final String snpName;
+
+        /** Snapshot directory path. */
+        @Nullable protected final String snpPath;
+
+        /** Consistent ID. */
+        protected final String consId;
+
+        /** Set of cache groups to be checked in the snapshot or {@code empty} to check everything. */
```

Comment:
Can we simplify the logic? If {@code null} then check everything, otherwise check the specified groups. I suppose Ignite already have check that the param must not be empty.